backend/tests/unit/services/test_data_masking_service.py

```
@@ -13,31 +13,31 @@
     """Test service initialization and builtin pattern loading."""
     
-    def test_initialization_without_registry(self):
-        """Test initialization with no registry disables all masking."""
-        service = DataMaskingService()
-        
-        assert service.mcp_registry is None
+    @pytest.mark.parametrize("registry,expected_registry,expected_patterns", [
+        (None, None, ["api_key", "password"]),  # No registry
+        (Mock(), "mock", ["api_key", "password"]),  # With registry
+    ])
+    def test_initialization_scenarios(self, registry, expected_registry, expected_patterns):
```

Ruff (0.12.2)

Unused method argument: `expected_registry`

(ARG002)